### DISCUSSION TITLE
Is possible to create a VectorData where each element is a softlink?

### DISCUSSION THREAD

**Discussion by borrepp** [NeurodataWithoutBorders/helpdesk#49] [Q&A] [answered]
Hello, I'm wondering if I can use a vector of softlinks as an input to a DynamicTable or is better to use an "ObjectView"

**Comment by lawrence-mbf** [comment]
Hi @borrepp 

Are you using one of the scripts in MatNWB to generate a DynamicTable or are you creating it from scratch? Are you following a particular tutorial?

**Reply by borrepp** [reply]
Hello @lawrence-mbf 

I'm using Python. I'm trying to create a LabMetaData extension (following this tutorial https://nwb-overview.readthedocs.io/en/latest/extensions_tutorial/extension_examples/labmetadata_extension.html) to include some monkey task parameters. My question is, at the moment of creating the behavior-trials group (as a DynamicTable), can I make a reference to some parameters that I will store in LabMetaData? I can give you a more concrete example of what I want to do. In a given trial, the monkey has to choose between 2 visual cues. These visual cues came from a pool of 8 possible stimuli (triangles of different sizes, colors, and locations). I'm planning to create DynamicTable in a group called VisualCues in the LabMetada extension and I was wondering if within the behavior-trials table is possible to reference (via soflink or objectview, which I used a long time ago in MatNWB) the information stored in the LabMetada Let's say, in the behavior-trials I will have a "column" called "choiceTarget1_ID" and the input will refer to one row in LabMetadata. Does it make sense? Another option is to create multiple "columns" in the behavior trials (i.e. "choiceTarget1_Shape", "choiceTarget1_Color', etc.)

Sorry for the long answer. Hopefully, I didn't make it more confusing. 

Thanks in advance.

**Reply by lawrence-mbf** [reply]
Oops, with the ObjectView talk I assumed MatNWB. @rly `ObjectView` are just Object References in HDF5, do soft links work there?

**Reply by lawrence-mbf** [reply, marked as answer]
@borrepp So you should be able to pass a vector of container objects which will automatically resolve themselves to `ObjectView` on write. Was there a particular reason why you wish to use soft links? Soft links will also not work here since they cannot be written to datasets.

**Reply by borrepp** [reply]
Hi @lawrence-mbf, I didn't know what to use. I'm just starting to use NWB and I was wondering which should be the correct way. I will use a vector of container objects. Thanks¡

**Reply by rly** [reply]
If you are using PyNWB, here is a quick example using generic object types TimeSeries and DynamicTable (but please use NWB data types that are specific to your data): https://gist.github.com/rly/fcea32480f0444d704b0aa453e63783c

**Reply by borrepp** [reply]
Thanks @rly ¡¡¡ I will check the example.
